develop/tutorials/articles/100-tooling/04-liferay-workspace/06-validating-modules-against-the-target-platform.markdown

```diff
@@ -0,0 +1,236 @@
+[Google Guava](https://opensource.google.com/projects/guava))?. The `resolve`
+task fails by default if your project depends on this project type. You
+probably plan to have this project deployed and available in @product@ at
+runtime, so it's not a concern, but the resolver doesn't know that; you'll need
+to customize the resolver to bypass this.
+
+There are three ways you can do this:
+
+- [Embed the third party library in your module](#embed-the-third-party-project-in-your-module)
+- [Add the third party library's capabilities to the current static set of resolver capabilities](#add-the-third-party-librarys-capabilities-to-the-current-static-set-of-resolver-capabilities)
+- [Skip the resolving process for your module](#skip-the-resolving-process-for-your-module)
+
+For help resolving third party dependency errors, see the
+[Resolving Third Party Library Package Dependencies](/develop/tutorials/-/knowledge_base/7-1/adding-third-party-libraries-to-a-module)
+tutorial.
+
+#### Embed the Third Party Library in Your Module
+
+If you only have one module that depends on the third party project, you can
+bypass the resolver failure by embedding the JAR in your module. This is not
+best practice if more than one project in your @product@ instance depends on
+that module. See the
+[Embedding Libraries in a Module](/develop/tutorials/-/knowledge_base/7-1/adding-third-party-libraries-to-a-module#embedding-libraries-in-a-module)
+section for more details.
+
+#### Add the Third Party Library's Capabilities to the Current Static Set of Resolver Capabilities
+
+You can add your third party dependencies to the distro JAR's list of
+capabilities by listing them as provided modules. Do this by adding the
+following Gradle code into your workspace's root `build.gradle` file:
+
+    dependencies {
+        providedModules group: "GROUP_ID", name: "NAME", version: "VERSION"
+    }
+
+For example, if you wanted to provide
+[Google Guava](https://opensource.google.com/projects/guava) as a provided
+module, it would look like this:
+
+    dependencies {
+        providedModules group: "com.google.guava", name: "guava", version: "23.0"
+    }
+
+Not only does this provide the third party dependency to the resolver, it also
+downloads and includes it in your @product@ bundle's `osgi/modules` folder when
+you initialize it (e.g., `gradlew initBundle`).
+
+#### Skip the Resolving Process for Your Module
+
+It may be easiest to skip validating your module during the resolve process. To
+do this, open your workspace's root `build.gradle` file and insert the following
+Gradle code at the bottom of the file:
+
+    targetPlatform {
+        resolveOnlyIf { project ->
+            project.name != 'PROJECT_NAME'
+        }
+    }
+
+Be sure to replace the `PROJECT_NAME` filler with your module's name (e.g.,
+`test-api`).
+
+Now the `resolve` task skips your module project.
+
+### Depending on a Platform That Must Be Installed in @product@
```

Comment:
Lets rename this to something about a "custom" distribution of Liferay (you've modified one of the default bundles).

Reply:
@gamerson could this really be two scenarios for generating a custom distro JAR?

1. Depending on a Platform not included in Liferay by default
2. Depending on a customized distribution of Liferay

Or are these more of the same thing?